src/query/block/scalar.go

```diff
@@ -0,0 +1,172 @@
+	}
+
+	return &Result{
+		Blocks: blocks,
+	}
+}
+
+// NewScalarBlock creates a scalar block containing val over the bounds
+func NewScalarBlock(val float64, bounds Bounds) Block {
+	return &ScalarBlock{
+		val: val,
+		meta: Metadata{
+			Bounds: bounds,
+			Tags:   make(models.Tags),
+		},
+	}
+}
+
+// StepIter returns a StepIterator
+func (b *ScalarBlock) StepIter() (StepIter, error) {
+	bounds := b.meta.Bounds
+	steps := bounds.Steps()
+	return &scalarStepIter{
+		meta: b.meta,
+		step: scalarStep{
+			vals: []float64{b.val},
+		},
+		numVals: steps,
+		idx:     -1,
+	}, nil
+}
+
+// SeriesIter returns a SeriesIterator
+func (b *ScalarBlock) SeriesIter() (SeriesIter, error) {
+	bounds := b.meta.Bounds
+	steps := bounds.Steps()
+	vals := make([]float64, steps)
+	for i := range vals {
+		vals[i] = b.val
+	}
+	return &scalarSeriesIter{
+		meta: b.meta,
+		vals: vals,
+		idx:  -1,
+	}, nil
+}
+
+// Close closes the scalar block
+func (b *ScalarBlock) Close() error { return nil }
+
+// Value returns the value for the scalar block
+func (b *ScalarBlock) Value() float64 { return b.val }
+
+type scalarStepIter struct {
+	meta    Metadata
+	step    scalarStep
+	numVals int
+	idx     int
+}
+
+// assert that scalarStepIter implements StepIter
+var _ StepIter = (*scalarStepIter)(nil)
+
+var seriesMeta = []SeriesMeta{
+	SeriesMeta{
+		Tags: models.Tags{},
+	},
+}
+
+func (it *scalarStepIter) Close()                   { /* No-op*/ }
+func (it *scalarStepIter) StepCount() int           { return it.numVals }
+func (it *scalarStepIter) SeriesMeta() []SeriesMeta { return seriesMeta }
+func (it *scalarStepIter) Meta() Metadata           { return it.meta }
+
+func (it *scalarStepIter) Next() bool {
+	it.idx++
+	bounds := it.meta.Bounds
+	it.step.time = bounds.Start.Add(time.Duration(it.idx) * bounds.StepSize)
+	return it.idx < it.numVals
+}
+
+func (it *scalarStepIter) Current() (Step, error) {
+	if it.idx >= it.numVals || it.idx < 0 {
+		return nil, fmt.Errorf("invalid scalar index: %d, numVals: %d", it.idx, it.numVals)
+	}
+	return &it.step, nil
+}
+
+type scalarStep struct {
+	vals []float64
+	time time.Time
+}
+
+// assert that scalarStepIter implements StepIter
+var _ Step = (*scalarStep)(nil)
+
+func (it *scalarStep) Time() time.Time   { return it.time }
+func (it *scalarStep) Values() []float64 { return it.vals }
+
+type scalarSeriesIter struct {
+	meta Metadata
+	vals []float64
+	idx  int
+}
+
+// assert that scalarStepIter implements StepIter
+var _ SeriesIter = (*scalarSeriesIter)(nil)
+
+func (it *scalarSeriesIter) Close()                   { /* No-op*/ }
+func (it *scalarSeriesIter) SeriesCount() int         { return 1 }
+func (it *scalarSeriesIter) SeriesMeta() []SeriesMeta { return seriesMeta }
+func (it *scalarSeriesIter) Meta() Metadata           { return it.meta }
+func (it *scalarSeriesIter) Next() bool {
+	it.idx++
+	return it.idx == 0
+}
+
+func (it *scalarSeriesIter) Current() (Series, error) {
+	if it.idx != 0 {
```

Comment:
we haven't traditionally been checking this in Current() usually rely on people to call Next() before Current

Reply:
I agree in general, but since this doesn't do any real "bounds" checking and just returns a value, the alternative is that this `panics` in this check, as it really shouldn't be returning anything if Next() is false.

I prefer to error than to panic here, but if you have particular preference towards panicing I can change to that

Reply:
same reply as above